### ISSUE TITLE
[Bug]: Satisfactory timeout values not working from defaults

### ISSUE THREAD

**Issue by evakq8r** [parkervcp/eggs#1707]
### Panel Version

1.8.1

### Wings Version

1.6.4

### Service

satisfactory

### Modified

Yes, I modified the egg (will provide details below)

### Expected Behavior

Start server, log into Satisfactory on Steam from PC, create world in Server Manager, join world, explore.

### Actual Behavior

Start server, log into Satisfactory on Steam from PC, create world, attempt to join world, timeout as connection took > 30 seconds. Configuration in Panel startup has been adjusted with values greater than 30 seconds, Steam client reports a timeout trying to connect to the server.

Server Panel config:

![image](https://user-images.githubusercontent.com/14250328/174021748-3d2e79e4-3dd1-4b67-9faa-d4a3c8a4f0ee.png)

Server Panel stats:

![image](https://user-images.githubusercontent.com/14250328/174022944-3edf3a8a-d36a-4733-97c7-489f7c630738.png)

Confirmed settings have been set in the Engine.ini file of the Server host:

```
root@<server>: cat /mnt/cache/pterodactyl-node/data/<UUID_of_container>/FactoryGame/Saved/Config/LinuxServer/Engine.ini 
[/Script/FactoryGame.FGSaveSession]
mNumRotatingAutosaves=3

[CrashReportClient]
bImplicitSend=true

[/Script/OnlineSubsystemUtils.IpNetDriver]
InitialConnectTimeout=300
ConnectionTimeout=200
```
Steam error:

![2022-06-16 17_20_10-Satisfactory](https://user-images.githubusercontent.com/14250328/174022031-d73a6994-f019-45e5-a9d0-2cd6b92011c6.png)

### Steps To Reproduce

Step 1: Import egg from https://github.com/parkervcp/eggs/blob/master/game_eggs/steamcmd_servers/satisfactory/egg-satisfactory.json
Step 2: Create server with required parameters (only changes made were ports (moved the defaults to +1 as I already have a Satisfactory docker running from ich777 that does **not** experience this issue.
Step 3: Wait for daemon to indicate server install was successful.
Step 4: Load Satisfactory, open Server Manager
Step 5: Configure new server with name and password
Step 6: Create world
Step 7: Wait for world to be created, attempt to join
Step 8: Review console logs for daemon. Daemon shows the world is setting up as normal, however Steam client on PC times out after 30 seconds to indicate a timeout occurred. Daemon logs suggest the loadtime for the environment was ~50 seconds.

NOTE: Issue does not occur with ich777/steamcmd:satisfactory docker container.

NOTE2: Wasn't able to get the pteropaste log line to work, used Pastebin instead with the same log file info.

### Install logs

https://pastebin.com/28U90VDM

**Comment by gOOvER** [comment]
Experimental?

**Comment by evakq8r** [comment]
Yes experimental, sorry. Internal IP was for testing. WAN IP experiences the same behaviour.

**Comment by gOOvER** [comment]
Experimental have a lot of Problems. Update to lastest Version.

btw: as Long its Experimental, you should Report such Errors to satisfactory bugtacker.

**Comment by evakq8r** [comment]
I'd agree if all docker containers experience the same problems, but only pterodactyl has the problem (as mentioned in the initial report).

Game is already latest* version as the game is set to auto-update.

**Comment by gOOvER** [comment]
The timeout is a wrong configurated Port. Bad the same Problem Yesterday and fix it with using other ports

**Comment by evakq8r** [comment]
The game is designed to work on multiple ports. The configuration in Pterodactyl was set up with all required ports and moved +1 (7777, 15000 and 15777 to 7778, 15001 and 15778 respectively), per the games wiki.

Moving the game back to the original ports **did not fix the issue**, as the exact same problems occur. Same steps to reproduce, same outcome.

I'll move back to the original docker container I was using as based on your responses, you do not appear to be willing to look at the problem and place blame on anything else but.

**Comment by gOOvER** [comment]
for me it works.

**Comment by evakq8r** [comment]
@lilkingjr1 

> What did you modify? Just default values? Or did you actually modify the egg itself? (not just parameter values)

Yes, just parameters, no actual egg modification (there was no option in the template for parameter modification, so selected yes on the egg).

> What port did you try to connect with in the client? This seems to imply 7778, but I could be wrong.

15778, as this is the server query port. The Server Manager ingame defaults to 15777 when trying to connect to a server, so I changed accordingly. That said, I also tried 7778 with no change in result.

> Source? I can't find anything saying +1 is okay, but I might have missed it. All I know is that from testing and according to the egg's [README](https://github.com/parkervcp/eggs/tree/master/game_eggs/steamcmd_servers/satisfactory#server-ports) that +1 will not work and cause connection issues.

[Here](https://satisfactory.fandom.com/wiki/Dedicated_servers#Manual_-_Command_line). Server query port can be changed freely, however it notes with the other two parameters **"If this port is already in use, the server will step up to the next port until an available one is found".** Docker doesn't work that dynamically to my knowledge, which is why I set the config to use a +1 port. Admittedly, it appears I overlooked that section of the README, so my apologies on that.

> You mean the same ports as your other server? Yeh, that definitely wont work.

I had turned off the other Satisfactory container I was running in order to use the ports, so it had worked but did not change the end result.

> try ports 7877, 15100, and 15877, and connect with the client via 15877.

I will try that shortly.

> Edit: Oh, also, `InitialConnectTimeout` and `ConnectionTimeout` only apply to slow _**but successful(ish)**_ connections. This would be were the client thinks it's connecting, but the server doesn't agree and drops the client after these times. If the client is connecting to the wrong port or a bad port, no connection will be made at all, so the server cannot drop any connection; the client itself will make that call, which sounds to be hard coded to ~30 sec. I believe that is the case here.

Hmm, I'm not sure I agree (yet) but I will take your word for it based on the fact the logs in the daemon do not error out to indicate a timeout had occurred, as only the Steam client does. At the time the timeout occurs, the daemon is still trying to load the world that has been created (alot of reference to async flushing, will grab a full log later if need be).

**Comment by evakq8r** [comment]
@lilkingjr1 

Changing the ports had no change to the outcome.

To rule out my own hardware being the problem, I rented access to a server from one of Pterodactyl's sponsors (https://aussieserverhosts.com/ - ASH) and experienced **the exact same issues**. The server ports used by ASH were different to the default (10450 - 10452 in my example), which proves the ports are not the problem.

From my perspective, Pterodactyl and hosting Satisfactory are not currently fit for purpose so I will revert to using the docker container that I have been previously that does not experience this issue.
